### PR TITLE
QA: Refactor test framework to get rid of Rubocop exceptions

### PR DESCRIPTION
**Pull request by srbarrios**
## What does this PR change?

Related to https://github.com/SUSE/spacewalk/issues/14192

Refactor the test framework to get rid of the Rubocop TO-DO list. We now have some tweaks in the rules, so we adapt Rubopcop to our style and we have an empty TO-DO list.

Main changes:
- Update Rubocop version to the last supported version for Ruby2.5
- Get empty the TO-DO list of Rubocop exceptions
- Add descriptions in all our methods
- Re-organize in a new folder all the XMLRPC API handlers
- Wrap all our Ruby methods inside different Ruby modules
- Move some methods defined inside step-definitions to our modules
- Remove un-used unit-tests, keeping only the checker of scenario names in a folder renamed tools
- Remove Travis configuration, as we don't use it anymore.
- Move all Cucumber Hooks into a new module Hooks
- Add a kind of Exception in all our raise calls (that could be re-visited after this PR, to have more appropriated types or define news)
- Define fixed versions for all our RubyGems, as required by Rubocop.
- Add a new Cop to check the existence of the Copyright Notice.
- Move all in-line comments to the above line of code.
- Some auto-corrections made by Rubocop regarding parenthesis and so on.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed

- [x] **DONE**

## Test coverage
- Cucumber tests were changed

- [x] **DONE**

## Links

Ports:
- Manager-4.1
- Manager-4.2

- [ ] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [x] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: testsuite/features/step_definitions/navigation_steps.rb [hunk 1]
```diff
@@ -393,20 +406,16 @@
...
 
 # access the clients
 Given(/^I am on the Systems overview page of this "([^"]*)"$/) do |host|
   system_name = get_system_name(host)
-  steps %(
-    Given I am on the Systems page
-    When I follow "#{system_name}"
-    And I wait until I see "System Status" text
-  )
+  steps %( Given I am on the Systems page When I follow "#{system_name}" And I wait until I see "System Status" text )
```
Comment (Bischoff) [inline testsuite/features/step_definitions/navigation_steps.rb:418]: this looks like an error

File: testsuite/features/support/xmlrpc/xmlrpc_powermanagement.rb [hunk 2]
```diff
@@ -1,55 +1,41 @@
-# Copyright (c) 2020 SUSE LLC.
+# Copyright (c) 2021 SUSE LLC.
```
Comment (Bischoff) [inline testsuite/features/support/xmlrpc/xmlrpc_powermanagement.rb:1]: 2020-2021

File: testsuite/features/support/custom_formatter.rb [hunk 3]
```diff
@@ -1,23 +1,28 @@
 # Copyright (c) 2019 SUSE LLC
```
Comment (Bischoff) [inline testsuite/features/support/custom_formatter.rb:1]: 2019-2021

File: testsuite/features/support/pretty_formatter.rb [hunk 4]
```diff
@@ -1,8 +1,8 @@
-# Copyright (c) 2013-2019 SUSE LLC.
+# Copyright (c) 2021 SUSE LLC.
```
Comment (Bischoff) [inline testsuite/features/support/pretty_formatter.rb:1]: 2013-2021, no ?

File: testsuite/features/support/xmlrpc/xmlrpc_configchannel.rb [hunk 5]
```diff
@@ -1,51 +1,63 @@
-# Copyright (c) 2020 SUSE LLC.
+# Copyright (c) 2021 SUSE LLC.
```
Comment (Bischoff) [inline testsuite/features/support/xmlrpc/xmlrpc_configchannel.rb:1]: 2020-2021

File: testsuite/features/step_definitions/common_steps.rb [hunk 6]
```diff
@@ -365,10 +385,7 @@
 
 When(/^I enter the SCC credentials$/) do
   scc_username, scc_password = ENV['SCC_CREDENTIALS'].split('|')
-  steps %(
-    And I enter "#{scc_username}" as "edit-user"
-    And I enter "#{scc_password}" as "edit-password"
-  )
+  steps %( And I enter "#{scc_username}" as "edit-user" And I enter "#{scc_password}" as "edit-password" )
```
Comment (Bischoff) [inline testsuite/features/step_definitions/common_steps.rb:388]: this looks like an error

File: testsuite/features/step_definitions/common_steps.rb [hunk 7]
```diff
@@ -1110,13 +1149,12 @@ def server_secret
...
+
     puts tr.text
     tr.find("##{package}-pkg-state").select('Removed')
     next if has_css?('#save[disabled]')
-    steps %(
-      Then I click on "Save"
-      And I click on "Apply"
-    )
+
+    steps %( Then I click on "Save" And I click on "Apply" )
```
Comment (Bischoff) [inline testsuite/features/step_definitions/common_steps.rb:1157]: this looks like an error (it's unreadable in any case)

**Review by Bischoff — CHANGES_REQUESTED** [review]
see remarks

File: testsuite/features/support/pretty_formatter.rb (hunk 4 above)
Comment (srbarrios) [reply testsuite/features/support/pretty_formatter.rb:1]: Actually  I created it in 2019. So 2019-2021

**Comment by srbarrios** [comment]
Testing core here https://ci.suse.de/view/Manager/view/Uyuni/job/uyuni-prs-ci-tests/386/console

**Pull request merged**